bundles/org.openhab.binding.yioremote/src/main/java/org/openhab/binding/yioremote/internal/YIOremoteDockHandler.java

```diff
@@ -0,0 +1,380 @@
+    private @Nullable URI websocketAddress;
+    private YioRemoteDockHandleStatus yioRemoteDockActualStatus = YioRemoteDockHandleStatus.UNINITIALIZED_STATE;
+    private @Nullable Future<?> authenticationjob;
+    private @Nullable Future<?> websocketpollingjob;
+    public String receivedmessage = "";
+    private JsonObject recievedjson = new JsonObject();
+    private boolean heartbeat = false;
+    private boolean authenticationok = false;
+    private String receivedstatus = "";
+    private String lastsendircode = "";
+    private IRCode ircodereceivedhandler = new IRCode();
+    private IRCode ircodesendhandler = new IRCode();
+    private IRCodeSendMessage ircodesendmessagehandler = new IRCodeSendMessage(ircodesendhandler);
+    private AuthenticationMessage authenticationmessagehandler = new AuthenticationMessage();
+    private IRReceiverMessage irreceivermessagehandler = new IRReceiverMessage();
+
+    public YIOremoteDockHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+        scheduler.execute(() -> {
+            try {
+                websocketAddress = new URI("ws://" + localConfig.host + ":946");
+                yioRemoteDockActualStatus = YioRemoteDockHandleStatus.AUTHENTICATION_PROCESS;
+            } catch (URISyntaxException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR,
+                        "Initialize web socket failed: " + e.getMessage());
+            }
+
+            yioremoteDockwebSocketClient.addMessageHandler(new YIOremoteDockWebsocketInterface() {
+
+                @Override
+                public void onConnect(boolean connected) {
+                    if (connected) {
+                        yioRemoteDockActualStatus = YioRemoteDockHandleStatus.CONNECTION_ESTABLISHED;
+                    } else {
+                        yioRemoteDockActualStatus = YioRemoteDockHandleStatus.CONNECTION_FAILED;
+                    }
+                }
+
+                @Override
+                public void onMessage(String message) {
+                    receivedmessage = message;
+                    logger.debug("Message recieved {}", message);
+                    recievedjson = convertStringtoJsonObject(receivedmessage);
+                    if (recievedjson.size() > 0) {
+                        if (decodeReceivedMessage(recievedjson)) {
+                            triggerChannel(getChannelUuid(GROUP_OUTPUT, STATUS_STRING_CHANNEL));
+                            updateChannelString(GROUP_OUTPUT, STATUS_STRING_CHANNEL, receivedstatus);
+                            if (yioRemoteDockActualStatus.equals(YioRemoteDockHandleStatus.CONNECTION_ESTABLISHED)
+                                    || yioRemoteDockActualStatus
+                                            .equals(YioRemoteDockHandleStatus.AUTHENTICATION_PROCESS)) {
```

Comment:
I overlooked that during the first reviews. Sorry for that. Please check all other Enum comparisons in all classes.
```suggestion
                            if (yioremotedockactualstatus == YioRemoteDockHandleStatus.CONNECTION_ESTABLISHED
                                    || yioremotedockactualstatus == YioRemoteDockHandleStatus.AUTHENTICATION_PROCESS) {
```

Reply:
done